### PR TITLE
Done - CSVTK tools

### PR DESCRIPTION
**Pull request by DarianHole**
## Tool
Adding the [csv toolkit tools](https://bioinf.shenwei.me/csvtk/) to galaxy

### What
Most of the csvtk tools have been added in individual wrappers connected by a macros file for as many inputs as possible. 

If there is one I missed that seems super useful let me know

Also sorry for whoever has to look at this PR, its gotten very large

### CODE REVIEW

**Comment by DarianHole** [comment]
My planemo broke so I am hoping that I wrote the tests well as I couldn't test them

File: tools/csvtk/awklike-filter.xml [hunk 1]
```diff
@@ -0,0 +1,183 @@
+<tool id="csvtk-awklike-filter" name="csvtk-advanced-filter" version="@VERSION@+@GALAXY_VERSION@">
+    <description> rows by awk-like artithmetic/string expressions</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements" />
```
Comment (bgruening) [inline tools/csvtk/awklike-filter.xml:6]: version_command could be included

File: tools/csvtk/awklike-filter.xml [hunk 2]
```diff
@@ -0,0 +1,183 @@
+<tool id="csvtk-awklike-filter" name="csvtk-advanced-filter" version="@VERSION@+@GALAXY_VERSION@">
```
Comment (bgruening) [inline tools/csvtk/awklike-filter.xml:1]: `_` is preferred for IDs

File: tools/csvtk/collapse.xml [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+            </when>
+            <when value="string">
+                <param type="text" name="in_text"
+                    optional="false"
+                    label="Input column name(s)"
+                    help="Specify columns to be used as keys for collapse. 
+                    Multiple columns can be given if separated by a ' , '. 
+                    Column numbers can be given too - ex. '1,2' will target columns 1 and 2. 
+                    Please see the help section below for more detailed info"
+                    />
```
Comment (bgruening) [inline tools/csvtk/collapse.xml:70]: this also needs a sanitizer ... single quotes should not be allowed for example.

File: tools/csvtk/collapse.xml [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+            <when value="column">
+                <param type="data_column" name="in_text"
+                        data_ref="in_1"
+                        multiple="false"
+                        label="Input column number"
+                        help="Select column to collapse on"
+                        />
+            </when>
+            <when value="string">
+                <param type="text" name="in_text"
```
Comment (bgruening) [inline tools/csvtk/collapse.xml:87]: That could be replaced by the stuff that you already have in your macros, left? A sanitizer would also be cool here.

Comment (DarianHole) [reply tools/csvtk/collapse.xml:65]: Thanks! Added the sanitizer to all of the text inputs

File: tools/csvtk/awklike-filter.xml (hunk 2 above)
Comment (DarianHole) [reply tools/csvtk/awklike-filter.xml]: Good to know

File: tools/csvtk/awklike-filter.xml (hunk 1 above)
Comment (DarianHole) [reply tools/csvtk/awklike-filter.xml:6]: Added to all tools, thanks!

**Pull request merged**